src/main/java/uk/ac/ebi/ampt2d/metadata/properties/SwaggerApiInfoProperties.java

```diff
@@ -0,0 +1,104 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.properties;
+
+import org.hibernate.validator.constraints.NotBlank;
+import org.hibernate.validator.constraints.URL;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+import javax.validation.Valid;
+
+@ConfigurationProperties(prefix="swagger.apiinfo")
+@Validated
+public class SwaggerApiInfoProperties {
+
+    @NotBlank(message="API title can not be blank.")
+    private String title;
+
+    private String description;
+
+    @NotBlank(message="API version can not be blank.")
+    private String version;
+
+    @URL(message="Terms of Service URL is not valid.")
```

Comment:
Does this accept a blank URL? Judging by the test properties file, it seems so, but just to be sure.

Reply:
Yes, it will accept a blank URL. And the swagger-ui will not show the link to the terms of service if the URL is blank or not defined.
If it is not blank, then the value will be validated against @URL constraint.